internals/daemon/api_notices.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"regexp"
+
+	"github.com/canonical/x-go/strutil"
+
+	"github.com/canonical/pebble/internals/overlord/state"
+)
+
+// Ensure custom keys are in the form "domain.com/key" (but somewhat more restrictive).
+var customKeyRegexp = regexp.MustCompile(
+	`^[a-z0-9]+(-[a-z0-9]+)*(\.[a-z0-9]+(-[a-z0-9]+)*)+(/[a-z0-9]+(-[a-z0-9]+)*)+$`)
```

Comment:
Just for my own knowledge, what is the strategy you take on line length (when to break it) ?

Reply:
I usually wrap comment text at 80 chars, but for code it's fairly informal! "Wrap at around 100" is probably my strategy, which is about what fits on my screen (with two code windows side by side on a widescreen monitor). In this case it was about 120 chars unwrapped, and I couldn't see the end, so I wrapped it. :-)